### PR TITLE
Fixing admba

### PR DESCRIPTION
**Pull request by viktorpm**
**What is this PR**

- [x] Bug fix
- [ ] Addition of a new feature
- [ ] Other

**What does this PR do?**

Fixes admba meshes and annotations mismatch, implements pooch to handle atlas downloading

## How has this PR been tested?

Tested locally, and atlases were regenerated on the HPC

## Is this a breaking change?

No 

## Does this PR require an update to the documentation?

No 

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality (unit & integration)
- [ ] The documentation has been updated to reflect any changes
- [ ] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Comment by alessandrofelder** [comment]
Hey @viktorpm 
I'll let you continue on this after the various debugging sessions last week (thanks also to @sfmig for maths help, and to @niksirbi for helpful anatomical eyeballing :grin: )

I haven't changed all the resolutions in the config yet, but we have concluded that 
* for anisotropic atlases that don't come in ASR, we need to keep track of resolution in source space (to scale mesh points correctly) and in target space (to write output metadata correctly)
* the ADMBA are in RSA according to our convention
* as part of #219 we should generate the mesh points from the _transformed_ annotation stack (ie in target space)
* for now, we've implemented a hacky way of dealing with this that won't affect other atlases.

File: brainglobe_atlasapi/atlas_generation/atlas_scripts/admba_3d_dev_mouse.py [hunk 1]
```diff
@@ -314,8 +319,13 @@ def create_atlas(
         species="Mus musculus",
         atlas_link="https://search.kg.ebrains.eu/instances/8ab25629-bdac-47d0-bc86-6f3aa3885f29",
         atlas_file_url="https://data.kg.ebrains.eu/zip?container=https://object.cscs.ch/v1/AUTH_4791e0a3b3de43e2840fe46d9dc2b334/ext-d000023_3Drecon-ADMBA-E11pt5_pub",
-        orientation="lsa",
-        resolution=(16, 16, 20),
+        orientation="rsa",
```
Comment (alessandrofelder) [inline brainglobe_atlasapi/atlas_generation/atlas_scripts/admba_3d_dev_mouse.py:322]: This is the correct way around, others need to be adapted!

**Comment by adamltyson** [comment]
> the ADMBA are in RSA according to our convention

Isn't our convention [asr](https://github.com/brainglobe/brainglobe-atlasapi/issues/293)?

**Review by alessandrofelder — APPROVED** [review]
Amazing - thank you @viktorpm

**Pull request merged**